### PR TITLE
Avoid unused const variable/verbose_print warning

### PR DESCRIPTION
**Pull request by hfp**
* Almost false positive aka incorrect warning.
* Shall be fixed regardless.

### CODE REVIEW

**Comment by alazzaro** [comment]
@hfp no sure I understand your change. Originally that function didn't exist with HIP on AMD. I think it exists so we can remove the entire condition...

**Comment by alazzaro** [comment]
Correction, it still [not supported](https://docs.amd.com/projects/HIP/en/latest/doxygen/html/group___global_defs.html#ga02ceb1513c852c4dd1ecf3cc459fda70) with HIP on AMD.

But then the proper condition should include NVIDIA too, something like:
```
#if defined(__CUDA) || defined(__HIP_PLATFORM_NVCC__)
```
and you can directly remove the entire condition...

File: src/acc/cuda_hip/acc_dev.cpp [hunk 1]
```diff
@@ -21,8 +21,8 @@
 #include <math.h>
 
 // for debug purpose
-#if defined(__HIP_PLATFORM_NVCC__)
-static const int verbose_print = 1;
+#if defined(__HIP_PLATFORM_NVCC__) && !defined(ACC_DEV_PRINT_VERBOSE) && 0
+#  define ACC_DEV_PRINT_VERBOSE
```
Comment (alazzaro) [inline src/acc/cuda_hip/acc_dev.cpp:25]: I will remove this part entirely....

File: src/acc/cuda_hip/acc_dev.cpp [hunk 2]
```diff
@@ -49,10 +49,8 @@ extern "C" int c_dbcsr_acc_set_active_device(int device_id) {
   // establish context
   ACC_API_CALL(Free, (0));
 
-#if defined(__HIP_PLATFORM_NVCC__)
-  if (verbose_print) {
-    ACC_API_CALL(DeviceSetLimit, (ACC(LimitPrintfFifoSize), (size_t)1000000000));
-  }
+#if defined(__HIP_PLATFORM_NVCC__) && defined(ACC_DEV_PRINT_VERBOSE)
+  ACC_API_CALL(DeviceSetLimit, (ACC(LimitPrintfFifoSize), (size_t)1000000000));
```
Comment (alazzaro) [inline src/acc/cuda_hip/acc_dev.cpp:53]: Here I will add a single:
```
#if defined(__CUDA) || defined(__HIP_PLATFORM_NVCC__)
ACC_API_CALL(DeviceSetLimit, (ACC(LimitPrintfFifoSize), (size_t)1000000000));
#endif
```

**Review by alazzaro — COMMENTED** [review]
follow-up to my last comment, I think we can simplify the code even further...

File: src/acc/cuda_hip/acc_dev.cpp (hunk 2 above)
Comment (hfp) [reply src/acc/cuda_hip/acc_dev.cpp:53]: Thanks, will do it accordingly! I simply overlooked the whole condition was meant to actually call this function (I was somehow convinced it's debug-only code).

Comment (alazzaro) [reply src/acc/cuda_hip/acc_dev.cpp:48]: It comes to plan with only debugging, meaning to restrict the printf on the other kernels where we use the debugging (e.g. https://github.com/cp2k/dbcsr/blob/46c43eda9914f76cb059a816a9439a3eb651070b/src/acc/cuda_hip/acc_event.cpp#L22). Of course, we do set it to `0`.
I think the entire `verbose_print = 1` in this file is really misleading, so better to remove it. Thanks!

**Comment by hfp** [comment]
This PR seems to cause:

>   CUDA RUNTIME API error: DeviceSetLimit failed with error cudaErrorInvalidValue.

( tested on H100 device )

**Comment by alazzaro** [comment]
Is this only H100? Maybe something has changed there?

**Pull request merged**